lib/auth0/api/v2/guardian.rb

```diff
@@ -0,0 +1,145 @@
+
+          path = "#{guardian_enrollments_path}/#{enrollment_id}"
+          delete(path)
+        end
+
+        # Retrieves SMS enrollment and verification templates.
+        # @see https://auth0.com/docs/api/management/v2#!/Guardian/get_templates
+        # @return [json] Returns enrollment and verification templates.
+        def guardian_sms_templates
+          path = "#{guardian_factors_path}/sms/templates"
+          get(path)
+        end
+        alias get_guardian_sms_templates guardian_sms_templates
+
+        # Updates SMS enrollment and verification SMS templates.
+        # @see https://auth0.com/docs/api/management/v2#!/Guardian/put_templates
+        # @param body [hash] The parameters to update.
+        #
+        # @return [json] Returns updated SMS enrollment and verification templates.
+        def guardian_update_enrollment_verification_templates(body)
+          raise Auth0::InvalidParameter, 'Must supply a valid body' if body.to_s.empty? || body.empty?
+
+          path = "#{guardian_factors_path}/sms/templates"
+          patch(path, body)
+        end
+
+        # Retrieves provider configuration for AWS SNS.
+        # @see https://auth0.com/docs/api/management/v2#!/Guardian/get_sns
+        # @return [json] Returns provider configuration for AWS SNS.
+        def guardian_sns_provider_config
+          path = "#{guardian_factors_path}/push-notification/providers/sns"
+          get(path)
+        end
+        alias get_guardian_sns_provider_config guardian_sns_provider_config
+
+        # Updates provider configuration for AWS SNS.
+        # @see https://auth0.com/docs/api/management/v2#!/Guardian/put_sns
+        # @param body [hash] The parameters to update.
+        #
+        # @return [json] Returns updated provider configuration for AWS SNS.
+        def guardian_update_sns_provider_config(body)
+          raise Auth0::InvalidParameter, 'Must supply a valid body' if body.to_s.empty? || body.empty?
+
+          path = "#{guardian_factors_path}/push-notification/providers/sns"
+          patch(path, body)
+        end
+
+        # Retrieves provider configuration for Twilio.
+        # @see https://auth0.com/docs/api/management/v2#!/Guardian/get_twilio
+        # @return [json] Returns provider configuration for Twilio.
+        def guardian_twillo_provider_config
+          path = "#{guardian_factors_path}/sms/providers/twilio"
+          get(path)
+        end
+        alias get_guardian_twillo_provider_config guardian_twillo_provider_config
+
+        # Updates provider configuration for Twilio.
+        # @see https://auth0.com/docs/api/management/v2#!/Guardian/put_twilio
+        # @param body [hash] The parameters to update.
+        #
+        # @return [json] Returns updated provider configuration for Twilio.
+        def guardian_update_twillo_provider_config(body)
+          raise Auth0::InvalidParameter, 'Must supply a valid body' if body.to_s.empty? || body.empty?
```

Comment:
Are both `body.to_s.empty?` and `body.empty?` checks necessary? If so, should the order be swapped? This pertains to all of the ones below as well.

Reply:
I wasn't sure about that, taken this pattern from

https://github.com/certsy/ruby-auth0/blob/c51be2a3fa755a790cc807c4e30c05f6da5ef880/lib/auth0/api/v2/users.rb#L108

I believe it's redundant though, left `body.to_s.empty?` only.